x-pack/test/security_solution_cypress/cypress/support/es_client.ts

```diff
@@ -0,0 +1,152 @@
+  on: Cypress.PluginEvents,
+  config: Cypress.PluginConfigOptions
+): Promise<void> => {
+  const isServerless = config.env.IS_SERVERLESS;
+  const isCloudServerless = config.env.CLOUD_SERVERLESS;
+
+  const user = {
+    username: config.env.ELASTICSEARCH_USERNAME,
+    password: config.env.ELASTICSEARCH_PASSWORD,
+  };
+
+  const authOverride = isServerless ? (isCloudServerless ? user : systemIndicesSuperuser) : user;
+
+  const client = createEsClient({
+    url: config.env.ELASTICSEARCH_URL,
+    username: authOverride?.username,
+    password: authOverride?.password,
+  });
+
+  on('task', {
+    putMapping: async (index: string) => {
+      await client.indices.putMapping({
+        index,
+        body: {
+          dynamic: true,
+        },
+      });
+      return null;
+    },
+    bulkInsert: async (body) => {
+      await client.bulk({
+        refresh: true,
+        body,
+      });
+      return null;
+    },
+    refreshIndex: async (index: string) => {
+      try {
+        await client.indices.refresh({ index });
+        return true;
+      } catch (error) {
+        return false;
+      }
+    },
+    searchIndex: async (index) => {
+      try {
+        const response = await client.search({
+          index,
+          body: {
+            query: {
+              match_all: {},
+            },
+          },
+        });
+
+        return response.hits.hits.length;
+      } catch (error) {
+        return null;
+      }
+    },
+    deleteDataStream: async (dataStreamName) => {
+      try {
+        await client.indices.deleteDataStream({
+          name: dataStreamName,
+        });
+
+        return { success: true };
+      } catch (error) {
+        return { success: false, error: error.message };
+      }
+    },
+    deleteIndex: async (index) => {
+      try {
+        await client.indices.delete({
+          index,
+        });
+
+        return { success: true };
+      } catch (error) {
+        return { success: false, error: error.message };
+      }
+    },
+    deleteDocuments: async (index: string) => {
+      await client.deleteByQuery({
+        index,
+        body: {
+          query: {
+            match_all: {},
+          },
+        },
+        conflicts: 'proceed',
+        scroll_size: 10000,
+        refresh: true,
+      });
+      return null;
+    },
+    createIndex: async ({ index: indexName, properties }) => {
+      const result = await client.indices.create({
+        index: indexName,
+        body: {
+          mappings: {
+            properties,
+          },
+        },
+      });
+      return result;
+    },
+    createDocument: async ({ index: indexName, document }) => {
+      const result = await client.index({
+        index: indexName,
+        body: document,
+        refresh: 'wait_for',
+      });
+
+      return result;
+    },
+    deleteSecurityRulesFromKibana: async () => {
+      await client.deleteByQuery({
+        index: '.kibana_*',
+        body: {
```

Comment:
This is something the team that uses that method needs to decide. 

As this method is used just to clean-up the state and not to test any internal call, it is fine for me to use it as it is today, specially because we don't need to first get all the ids or references of the rules before deleting them.